src/settings.cpp

```diff
@@ -72,27 +72,28 @@ void SettingsBase::setSetting(std::string key, std::string value)
     }
 }
 
-std::string SettingsBase::getSettingString(std::string key)
+std::string SettingsBase::getSettingString(std::string key) const
 {
     if (setting_values.find(key) != setting_values.end())
     {
-        return setting_values[key];
+        return setting_values.at(key);
     }
     if (parent)
     {
         return parent->getSettingString(key);
     }
     
+		SettingsBase& nonConstThis = const_cast<SettingsBase&>(*this);
     if (SettingRegistry::getInstance()->settingExists(key))
     {
-        setting_values[key] = SettingRegistry::getInstance()->getSettingConfig(key)->getDefaultValue();
+        nonConstThis.setting_values[key] = SettingRegistry::getInstance()->getSettingConfig(key)->getDefaultValue();
```

Comment:
getSettingString is a "get" function and is surprising that it is not const. It is the foundation of all other getSetting* functions, which are interpreting data into a specific data type and not modifying the Settings. If these get functions cannot be made const, nothing else can.